python_modules/dagster/dagster/_utils/pydantic_yaml.py

```diff
@@ -0,0 +1,56 @@
+from typing import Type, TypeVar
+
+from pydantic import BaseModel, ValidationError
+
+from dagster._model.pydantic_compat_layer import build_validation_error
+
+from .source_position import KeyPath, populate_source_position_and_key_paths
+from .yaml_utils import parse_yaml_with_source_positions
+
+T = TypeVar("T", bound=BaseModel)
+
+
+def parse_yaml_file(cls: Type[T], src: str, filename: str = "<string>") -> T:
+    """Parse the YAML source and create a Pydantic model instance.
+    This function takes a YAML source string, a Pydantic model class, and an optional
+    filename, and returns an instance of the Pydantic model. It performs the following
+    steps:
+    1. Parses the YAML source with source position information using
+       `parse_yaml_with_source_positions()`.
+    2. Validates the parsed data against the provided Pydantic model class, using the
+       `parse_obj()` method.
+    3. Populates the `_source_position_and_key_path` attribute of the Pydantic model instance using
+       the source position information.
+
+    Args:
+        cls (type[T]): The Pydantic model class to use for validation.
+        src (str): The YAML source string to be parsed.
+        filename (str): The filename associated with the YAML source, used for error reporting.
+            Defaults to "<string>" if not provided.
+
+    Returns:
+        T: An instance of the Pydantic model class, with the `_source_position_and_key_path`
+            attribute populated on it and all the objects inside it.
+
+    Raises:
+        ValidationError: If the YAML data does not conform to the Pydantic model schema.
```

Comment:
Worth mentioning in docstr that we also implement custom error building logic here, that seems to be the bulk of what this fn actually does

Reply:
👍